### ISSUE TITLE
Allow to disable foreign key constraint for relations

### ISSUE THREAD

**Issue by Frando** [prisma/prisma#15744]
## Problem

I'm using Prisma with PostgreSQL. In out data model, dangling relations are allowed. We import data that may contain relations to other entities that do not exist in the database, but might be added later on. This is not possible with prisma, because all relations have a `REFERENCES` foreign key constraint forced.

## Suggested solution

An optional attribute or argument to the relation attribute that disables the foreign key constraint. Everything else works as it does currently. But you can add rows with values in the scalar column of a reference that do not exist in the referenced table. When joining or querying, NULL would be returned instead.

## Alternatives

None I know of apart from manually deleting the constraints with raw SQL after each migration.

## Additional context

In #8108 someone mentioned that for MySQL, foreign key constraints are disabled when switching to the Planetscale driver. However, this does not work of course when using PostgreSQL.

**Comment by janpio** [comment]
Just disabling foreign keys is possible via: https://www.prisma.io/docs/concepts/components/prisma-schema/relations/referential-integrity But if I understand correctly, your use case is a bit more nuanced?

**Comment by Frando** [comment]
Yes I will likely have to disable referential integrity entirely by switching to `prisma` for now. However I'd prefer to disable foreign key constraints only on some references and keep them for others.

**Comment by rcbevans** [comment]
It would be great to support unconstrained relations to facilitated a shortcut to effectively perform a join.

In my datamodel I have Users which have email PKs, and in a tenant Contacts which have unique emails per tenant contact.  I would love to be able to easily get a User corresponding to a Contact *WITHOUT* that creating a foreign key constraint because in my case, a Contact isn't necessarily also a User of the product.

In my schema I defined the relations as

```
model User {
email String @id()
...
contacts Contact[]
}

model Contact {
...
email String @unique
...
user User? @relation(fields: [email], references: [email], onDelete: NoAction)
```

But despite user being Nullable, which to me would suggest it's totally possible for an associated user to not exist, and therefore implies no FK constraint, Prisma is creating a constraint.

For now I'm having to work around it by making a secondary query of users with email in the set of contact emails, then match them up.

**Comment by mdesousa** [comment]
this would be very useful, see #28505

**Comment by zacaj** [comment]
Bump

Would be really handy to have a param on `@relation` to disable the constraint- having to disable this DB wide isn't really viable
